custom_components/octopus_energy/__init__.py

```diff
@@ -75,6 +76,9 @@ async def async_update_electricity_rates_data():
       """Fetch data from API endpoint."""
       # Only get data every half hour or if we don't have any data
       if (DATA_RATES not in hass.data[DOMAIN] or (utcnow().minute % 30) == 0 or len(hass.data[DOMAIN][DATA_RATES]) == 0):
+        is_smart_electricity_meter = False
+        if CONFIG_SMART_ELECTRICITY_METER in config:
```

Comment:
I prefer things coming from the API as much as possible. I believe if you add the following 

```
smartImportElectricityMeter {
  deviceId
}
```

at line 28 of `apiClient`.

The logic should be similar to how `smartExportElectricityMeter ` is handled to determine if it's a smart meter.